Apply versification on import

jsarabia opened the pull request:
Applies versification on importing new source, then updates content and verse bridges

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/721)
<!-- Reviewable:end -->

jsarabia committed:
Preallocate content from versification in NewSourceImporter

Preallocates Collection and Content entries from a versification file, and then updates the text and bridges verses after constructing an OtterTree from the text.

jsarabia committed:
Modify ProjectFilesAccessor to get content from a chapter with verse bridges

jsarabia committed:
Filter out bridged verses from verse list

AnonymousWalker commented:
*[`jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/ChapterPageViewModel.kt` line 406 at r1](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/721#-NQWsv1E4FDCuNlCLkp7:-NQWsv1FDZ_2vK1JMfHZ:bk10ml4) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/532f213f65571da64c93b3929fe0eb3192bf69f1/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/ChapterPageViewModel.kt#L406)):*
> ```Kotlin
>                 filteredContent.removeIf { card ->
>                     (card.chunkSource != null && card.chunkSource.draftNumber < 0) ||
>                         (card.chunkSource != null && card.chunkSource.bridged)
> ```

`card.chunkSource != null && (card.chunkSource.draftNumber < 0 || card.chunkSource.bridged)`


<!-- Sent from Reviewable.io -->

jsarabia reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/721)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @AnonymousWalker)

___
*[`jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/ChapterPageViewModel.kt` line 406 at r1](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/721#-NQWsv1E4FDCuNlCLkp7:-NQ_RQ6z0QWh1ENyARn8:bvtu5uf) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/532f213f65571da64c93b3929fe0eb3192bf69f1/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/ChapterPageViewModel.kt#L406)):*
<details><summary><i>Previously, AnonymousWalker (Tony Tran) wrote…</i></summary><blockquote>

`card.chunkSource != null && (card.chunkSource.draftNumber < 0 || card.chunkSource.bridged)`
</blockquote></details>

The compiler isn't understanding that and requires ?. on the second expression

![Screenshot 2023-03-15 at 9.25.40 AM.png](https://files.reviewableusercontent.io/images/reviewable/9613175/HUi4tbyIJegacy6oLtsM6a7m/Screenshot%202023-03-15%20at%209.25.40%20AM.png)


<!-- Sent from Reviewable.io -->

jsarabia committed:
Update ChapterPageViewModel.kt

jsarabia committed:
Versification Fix

Fixes upsert, which was not working previously

Adds ufw versification, which is the same as ulb versification.

jsarabia committed:
Provide a test versification repository over DI

jsarabia committed:
Adjust Text Counts

Text counts are slightly off for verses allocated by versification which do not have a verse in the ULB, ie 2co 13

jsarabia committed:
Skip Versification Preallocation on null or unknown versification

For containers that do not have a versification (obs) the preallocation and update is skipped, and the Tree created from the parse result of the content is used.

This is essential for things like OBS.

jsarabia committed:
Fix versification file

Revelation 12 and 3 John had incorrect verse counts, which are corrected.

As a result, the test count changes from a previous PR are reverted as these are now accounted for.

jsarabia committed:
Update obs-tn-biel-v6.zip

Remove versification from the manifest (which was incorrect)
